torch/csrc/distributed/c10d/CUDASymmetricMemoryOps.cu

```diff
@@ -484,6 +489,63 @@ at::Tensor two_shot_all_reduce_(
   return input;
 }
 
+at::Tensor memset32_(
+    at::Tensor& input,
+    int64_t offset,
+    int64_t val,
+    int64_t count) {
+#if !defined(USE_ROCM) && defined(PYTORCH_C10_DRIVER_API_SUPPORTED)
+  TORCH_CHECK(
+      input.dim() == 1 && input.is_contiguous() &&
+          input.scalar_type() == c10::ScalarType::UInt32,
+      "CUDASymmetricMemoryUtils::memset32: input must be "
+      "a flat, contiguous uint32 tensor.");
+
+  TORCH_CHECK(
+      offset > 0 && count > 0,
+      "CUDASymmetricMemoryUtils::memset32: "
+      "offset and count must be a positive integers.")
+
+  TORCH_CHECK(
+      val >= 0 &&
+          static_cast<size_t>(val) <= std::numeric_limits<uint32_t>::max(),
+      "CUDASymmetricMemoryUtils::memset32: "
+      "val must be in the range of [0, 4294967295] (uint32_t).")
+
+  auto element_size = c10::elementSize(input.scalar_type());
+  TORCH_CHECK(
+      offset + count <= input.numel(),
+      "CUDASymmetricMemoryUtils::memset32: offset + count (",
+      offset + count,
+      ") exceeded the numel of the input (",
+      input.numel(),
+      ")");
+
+  auto driver_api = c10::cuda::DriverAPI::get();
+  CUcontext pctx, prev;
+
+  c10::cuda::CUDAGuard guard(input.device());
+  C10_CUDA_DRIVER_CHECK(driver_api->cuDevicePrimaryCtxRetain_(
+      &pctx, (CUdevice)input.device().index()));
+
+  C10_CUDA_DRIVER_CHECK(driver_api->cuCtxGetCurrent_(&prev));
+  C10_CUDA_DRIVER_CHECK(driver_api->cuCtxSetCurrent_(pctx));
```

Comment:
This is really weird, we don't have to do anything of this sort in xFormers when invoking cuMemsetD32Async. This is literally how we do it: https://github.com/facebookresearch/xformers/blob/68b7fd14df5eb1d2558c52842b4206a14d2d20e9/xformers/csrc/sequence_parallel_fused/memset_32b_kernels.cu#L58-L77

Is it perhaps that we obtain a reference to the driver function by calling `cudaGetDriverEntryPoint`? (and this makes it work on the correct CUDA context?). I don't see any real other explanation...